src/auto-compounder/libraries/SlipstreamLogic.sol

```diff
@@ -0,0 +1,159 @@
+/**
+ * Created by Pragma Labs
+ * SPDX-License-Identifier: MIT
+ */
+pragma solidity 0.8.22;
+
+import { FixedPoint96 } from "../../../lib/accounts-v2/src/asset-modules/UniswapV3/libraries/FixedPoint96.sol";
+import { FixedPoint128 } from "../../../lib/accounts-v2/src/asset-modules/UniswapV3/libraries/FixedPoint128.sol";
+import { FixedPointMathLib } from "../../../lib/accounts-v2/lib/solmate/src/utils/FixedPointMathLib.sol";
+import { FullMath } from "../../../lib/accounts-v2/src/asset-modules/UniswapV3/libraries/FullMath.sol";
+import { ICLPool } from "../interfaces/Slipstream/ICLPool.sol";
+import { ISlipstreamPositionManager } from "../interfaces/Slipstream/ISlipstreamPositionManager.sol";
+import { PoolAddress } from "../../../lib/accounts-v2/src/asset-modules/Slipstream/libraries/PoolAddress.sol";
+import { TickMath } from "../../../lib/accounts-v2/src/asset-modules/UniswapV3/libraries/TickMath.sol";
+
+library SlipstreamLogic {
+    using FixedPointMathLib for uint256;
+
+    // The binary precision of sqrtPriceX96 squared.
+    uint256 internal constant Q192 = FixedPoint96.Q96 ** 2;
+
+    // The Slipstream Factory contract.
+    address internal constant CL_FACTORY = 0x5e7BB104d84c7CB9B682AaC2F3d509f5F406809A;
+
+    // The Slipstream NonfungiblePositionManager contract.
+    ISlipstreamPositionManager internal constant POSITION_MANAGER =
+        ISlipstreamPositionManager(0x827922686190790b37229fd06084350E74485b72);
+
+    /**
+     * @notice Calculates the underlying token amounts of accrued fees, both collected and uncollected.
+     * @param id The id of the Liquidity Position.
+     * @return amount0 The amount of fees in underlying token0 tokens.
+     * @return amount1 The amount of fees in underlying token1 tokens.
+     */
+    function _getFeeAmounts(uint256 id) internal view returns (uint256 amount0, uint256 amount1) {
```

Comment:
Since these are not used in the main compounder would move it to the viewer alone (already created a separate PR for this for uniswapV3Compounder